### ISSUE TITLE
GithubBrowserSample File Structure

### ISSUE THREAD

**Issue by feinstein** [googlesamples/android-architecture-components#407]
It would be very useful if you could provide an explanation for the file structure:

![image](https://user-images.githubusercontent.com/8120472/42137076-8591c674-7d3c-11e8-99ec-7dc3c5807caa.png)

For example, I don't know what `vo` stands for, or why there are adapters under `ui/common` and also under `binding`, which is outside from `ui`.

I would even put `db` inside `repository`, so it would be very useful to read about the logic behind the choosing of this file structure and have the opportunity to learn even more, on how to structure the architecture itself.

**Comment by MrChriZ** [comment]
Beyond this it would also be useful if each sample had a detailed explanation of what the sample is showing.
Even the basic sample has a lot going on and without knowing what it's trying to achieve beyond persistence it's hard to work out how it's achieving it.

**Comment by pedrovarela86** [comment]
v.o. stands for value object.

**Comment by feinstein** [comment]
Yeah, took me a while to figure it out, but eventually I did...but beyond that, I wanted to understand the reasoning behind the structure, why they prefer it this way and etc... So I could improve my architectural thinking
